js/rook.js

```diff
@@ -0,0 +1,59 @@
+import {Piece} from "./piece.js";
+
+export class Rook extends Piece {
+    /**
+     * Constructor: inherits from abstract Piece class
+     * @param id piece id
+     * @param rowPos current row position
+     * @param colPos current column position
+     * @param playerId player id
+     */
+    constructor(id, rowPos, colPos, playerId) {
+        super();
+        this.id = id;
+        this.rowPos = rowPos;
+        this.colPos = colPos;
+        this.playerId = playerId;
+        this.unicodeChar = "&#9814;";
+    }
+
+    /**
+     * Gets a list of possible moves the rook piece can make including attack moves
+     * @param board 8x8 board array
+     * @returns {*[]}
+     */
+    possibleMoves(board) {
+        let possibleMoves = [];
+        for (let i = this.rowPos + 1; i < 8; i++) { // check below
+            if (board[i][this.colPos] && board[i][this.colPos].getPlayerId() !== this.playerId) {
+                possibleMoves.push([i, this.colPos]);
+                break;
+            }
+            possibleMoves.push([i, this.colPos]);
```

Comment:
These `if` conditions are missing the case where you run into a piece with the same `playerId`; you'll want to break the loop in that case.

Reply:
good catch, thanks!